1 Niagara 4 Overview Exercise - BFile/myFile-rt/src/com/tridiumuniversity/myFile/BFile.java

```diff
@@ -0,0 +1,219 @@
+package com.tridiumuniversity.myFile;
+
+import javax.baja.nre.annotations.NiagaraAction;
+import javax.baja.nre.annotations.NiagaraProperty;
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.BComponent;
+import javax.baja.sys.*;
+@NiagaraType
+@NiagaraProperty(
+        name = "filename",
+        type = "String",
+        defaultValue = "myFile"
+)
+@NiagaraProperty(
+        name = "path",
+        type = "String",
+        defaultValue = "C:"
+)
+
+@NiagaraProperty(
+        name = "size",
+        type = "int",
+        defaultValue = "0.0"
+)
+
+@NiagaraProperty(
+        name = "createdDate",
+        type = "BAbsTime" ,
+        defaultValue = "BAbsTime.now()",
```

Comment:
You have to be careful when setting a `defaultValue` of `BAbsTime.now()`. Since Properties are `static`, `BAbsTime.now()` is only going to get called once for this class, not once for each instance of it (the actual call to `BAbsTime.now()` is on line 127). So every instance of `BFile` that gets added to your station is going to have the same default value for `createdDate`. For time values, I'd recommend using `BAbsTime.DEFAULT` like you did on line 35.